sdk/core/azure-core/azure/core/polling/base_polling.py

```diff
@@ -461,31 +625,133 @@ def from_continuation_token(cls, continuation_token: str, **kwargs) -> Tuple[Any
         initial_response.context.transport = client._pipeline._transport  # pylint: disable=protected-access
         return client, initial_response, deserialization_callback
 
-    def run(self):
+    def status(self) -> str:
+        """Return the current status as a string.
+
+        :rtype: str
+        :return: The current status.
+        """
+        if not self._operation:
+            raise ValueError("set_initial_status was never called. Did you give this instance to a poller?")
+        return self._status
+
+    def finished(self) -> bool:
+        """Is this polling finished?
+
+        :rtype: bool
+        :return: True if finished, False otherwise.
+        """
+        return _finished(self.status())
+
+    def resource(self) -> PollingReturnType_co:
+        """Return the built resource.
+
+        :rtype: any
+        :return: The built resource.
+        """
+        return self._parse_resource(self._pipeline_response)
```

Comment:
We're making the assumption here that `self._pipeline_response` is `PipelineResponseType`, but in the `initialize` method it seems to get set to `Any`?

Reply:
Let me re-read the code before replying on this one

Reply:
I re-typed them `PipelineResponseType`